src/topp/IDFilter.cpp

```diff
@@ -227,19 +223,16 @@ class TOPPIDFilter :
                                  "Similar to n_peptide_hits=1, but if there are ties between two or more highest scoring hits, none are kept.");
     registerStringOption_("best:n_to_m_peptide_hits", "[min]:[max]", ":", "Peptide hit rank range to extracts", false, true);
 
-    registerStringOption_("length", "[min]:[max]", ":", "Keep only peptide hits with a sequence length in this range.", false);
-
-    registerStringOption_("charge", "[min]:[max]", ":", "Keep only peptide hits with charge states in this range.", false);
 
     registerFlag_("var_mods", "Keep only peptide hits with variable modifications (as defined in the 'SearchParameters' section of the input file).", false);
 
-    registerFlag_("unique", "If a peptide hit occurs more than once per peptide ID, only one instance is kept.");
-    registerFlag_("unique_per_protein", "Only peptides matching exactly one protein are kept. Remember that isoforms count as different proteins!");
+    registerFlag_("remove_ambiguous_peptides", "If a peptide-spectrum match occurs more than once for a spectrum, only the first instance is kept.", true);
```

Comment:
```suggestion
    registerFlag_("remove_ambiguous_peptides", "Keep only the best PSM (peptide spectrum match) per spectrum and remove lower scoring matches.", true);
```

I think that is correct? actually the docu of the fxn is ambigous even...

Reply:
`@brief Removes duplicate peptide hits from each peptide identification, keeping only unique hits (per ID). By default, hits are considered duplicate if they compare as equal using PeptideHit::operator==. However, if @p seq_only is set, only the sequences (incl     . modifications) are compared. In both cases, the first occurrence of each hit in a peptide ID is kept, later ones are removed.`

So it seems it will only remove duplicated PSMs for a spectrum - and leaves the others. So you may still have a list of PSMs for the spectrum. - Is that correct? 

"Removes duplicated PSMs per spectrum and retrains the one with the higher score." (The one with the higher score is the one which occurs first - at least that is what I think.) 


There is another option which is called "keepNBestHits" which can be used to only use the best PSM per Spectrum.

Reply:
ok I see, so this is quite confusing especially with a flag called "ambigous peptides" - what is ambigous about that? Why are there even duplicates? maybe call the flag "remove_duplicate_psm" ?